testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/authz/EntitlementAPITest.java

```diff
@@ -233,6 +233,7 @@ public void testInvalidRequestWithClaimsFromPublicClient() throws IOException {
 
         try {
             getAuthzClient(AUTHZ_CLIENT_CONFIG).authorization(response.getAccessToken()).authorize(request);
+            fail("Should fail, public clients not allowed");
```

Comment:
We've been trying to avoid these constructs when testing expected exceptions and instead use the JUnit ExpectedException rule. See https://github.com/keycloak/keycloak/blob/master/wildfly/extensions/src/test/java/org/keycloak/vault/ElytronCSKeyStoreProviderTest.java as an example.